types/webpack-env/index.d.ts

```diff
@@ -68,6 +69,53 @@ declare namespace __WebpackModuleApi {
     }
     type ModuleId = string|number;
 
+    interface HotNotifierInfo {
+        type:
+          | 'self-declined'
+          | 'declined'
+          | 'unaccepted'
+          | 'accepted'
+          | 'disposed'
+          | 'accept-errored'
+          | 'self-accept-errored'
+          | 'self-accept-error-handler-errored';
+        /**
+         * The module in question.
+         */
+        moduleId: number;
+        /**
+         * For errors: the module id owning the accept handler.
+         */
+        dependencyId?: number;
+        /**
+         * For declined/accepted/unaccepted: the chain from where the update was propagated.
+         */
+        chain?: number[];
```

Comment:
Ideally, there should be a separate interface for each `type`, as the shape looks different. Optional properties don't make much sense in this case, as they are specific only to a subset of `type`s. If there is a need for a base/common type you can have a union type, with `type` as discriminant. E.g.:
```ts
interface HotNotifierInfoDeclined {
    type: "declined";
    chain: number[];
    //...
}
interface HotNotifierInfoAccepted {
    type: "accepted";
    outdatedModules: number[];
    // ...
}
type HotNotifierInfo = HotNotifierInfoDeclined | HotNotifierinfoAccepted | ....;
```

Reply:
@use-strict are you comfortable with the PR as-is?

Reply:
It's fine.